### PR TITLE
[RFC] Process feedback chapter 2

### PR DESCRIPTION
**Pull request by Balletie**

### CODE REVIEW

File: final/background/spoofax.tex [hunk 1]
```diff
@@ -8,13 +8,13 @@ \section{The Spoofax Language Workbench}
...
+that aspect that adheres to the specification. Any programming language can be
+divided into multiple distinct aspects. In \cref{fig:spoofax-relations}, four
 common aspects of programming languages are shown, along with the relation
 between the formal specification and implementation for each aspect. An example
-of an aspect is the \textit{syntax definition}: its formal specification is done
-with grammar rules, and its implementation consists of both a tokenizer and
-parser.
+of an aspect is the \textit{syntax}: its formal specification is done with
+grammar rules, and its implementation traditionally consists of both a tokenizer
+and parser.
```
Comment (Hjdskes) [inline final/background/spoofax.tex]: `a parser`?

File: final/background/spoofax.tex [hunk 2]
```diff
@@ -8,13 +8,13 @@ \section{The Spoofax Language Workbench}
...
-be divided into multiple distinct aspects. In \cref{fig:spoofax-relations}, four
+that aspect that adheres to the specification. Any programming language can be
+divided into multiple distinct aspects. In \cref{fig:spoofax-relations}, four
 common aspects of programming languages are shown, along with the relation
 between the formal specification and implementation for each aspect. An example
-of an aspect is the \textit{syntax definition}: its formal specification is done
-with grammar rules, and its implementation consists of both a tokenizer and
-parser.
+of an aspect is the \textit{syntax}: its formal specification is done with
+grammar rules, and its implementation traditionally consists of both a tokenizer
```
Comment (Hjdskes) [inline final/background/spoofax.tex]: Remove `both`.

File: final/background/spoofax.tex [hunk 3]
```diff
@@ -50,26 +50,24 @@ \section{The Spoofax Language Workbench}
...
-  well-formed AST can then be transformed, for example for desugaring or
-  optimization. This execution step falls under the aspect of \textit{term
-    rewriting} and can be formally specified by \textit{term rewrite rules}. For
-  the specification of this aspect, Spoofax provides a DSL called Stratego.
-
-  This aspect, too, is not discussed in this section. The relevant section
-  from the research report can be found in \cref{ssec:a-term-rewrite}.
+  This language aspect is not discussed in this section, but it is explained in
+  the research report in \cref{ssec:a-static-analysis}.
+\item \textbf{Program transformation:} A well-formed AST can then be
```
Comment (Hjdskes) [inline final/background/spoofax.tex:55]: Remove `then`.

File: final/background/spoofax.tex [hunk 4]
```diff
@@ -50,26 +50,24 @@ \section{The Spoofax Language Workbench}
...
+  transformed, for example for desugaring or optimization. This execution step
+  falls under the aspect of \textit{program transformation} and can be formally
+  specified by \textit{term rewrite rules}. For the specification of this
+  aspect, Spoofax provides a DSL called Stratego.
+
+  This aspect is also not discussed in this section, the relevant section from
```
Comment (Hjdskes) [inline final/background/spoofax.tex]: > section. The relevant ...

File: final/background/spoofax.tex [hunk 5]
```diff
@@ -81,10 +79,11 @@ \section{The Spoofax Language Workbench}
   \label{fig:spoofax-relations}
 \end{figure}
 
-This section concludes with a discussion on the other part of a language: its
-IDE. Spoofax provides IDE support by means of its Editor Services.
+After the explanation of the language aspects, this section concludes with a
```
Comment (Hjdskes) [inline final/background/spoofax.tex]: > After explaining the language aspects, ...

File: final/background/spoofax.tex [hunk 6]
```diff
@@ -81,10 +79,11 @@ \section{The Spoofax Language Workbench}
...
+discussion on the other part of a language: its IDE. Spoofax provides IDE
+support by means of its \textit{editor services}.
```
Comment (Hjdskes) [inline final/background/spoofax.tex]: `Editor Services`?

File: final/background/spoofax.tex [hunk 7]
```diff
@@ -123,9 +121,9 @@ \subsubsection{SDF3: syntax definition in Spoofax}
 \subsubsection{ASTs in Spoofax: Stratego terms}
 \label{sec:asts-spoof-strat}
 In Spoofax, an AST is represented as a Stratego term~\cite{Kats10a}. Terms in
-Spoofax come with a textual format, ATerms, which is similar to formats such as
-XML~\cite{Brand00}. As an example, a term representation of the arithmetic
-expression $2 + 2 - 4$ is shown in \cref{lst:aterm-example}.
+Spoofax come with a textual format called the ATerm Format~\cite{Brand00}. As an
```
Comment (Hjdskes) [inline final/background/spoofax.tex]: `comes`

Comment (Hjdskes) [inline final/background/spoofax.tex]: Is `Format` supposed to be capitalized?

File: final/background/spoofax.tex [hunk 8]
```diff
@@ -144,8 +142,8 @@ \subsection{Dynamic semantics}
 behaves~\cite{Winskel93}. There are many approaches to formally specify the
 dynamic semantics of a programming language (for an extensive treatment, see
 ``\usebibentry{Winskel93}{title}''~\cite{Winskel93}). One way, for example, is
-to specify the transformation of programs written in one programming language to
-programs of another language of which the dynamic semantics are already known.
+to specify the compilation to a different high-level language of which the
```
Comment (Hjdskes) [inline final/background/spoofax.tex]: It doesn't necessarily have to be higher level, does it? E.g. C -> assembly.

File: final/background/spoofax.tex [hunk 9]
```diff
@@ -199,8 +198,13 @@ \subsubsection{DynSem: rule-based dynamic semantics}
 components and \textit{read-write} semantic components. An environment is a
 read-only semantic component: a variable is bound only within a certain scope,
 and any other reduction rules that are applied outside of that scope do not see
-this environment. A store is a read-write semantic component: changes in the
-store are visible everywhere, as it represents mutable state.
+this environment. Read-only semantic components therefore are only propagated
+downwards into the premises: the changed read-only semantic component does not
+appear in the result of the premise, and therefore cannot be propagated again in
+the next premise. A store is a read-write semantic component: changes in the
+store are visible everywhere, as it represents mutable state. In contrast with a
```
Comment (Hjdskes) [inline final/background/spoofax.tex]: `In contrast to`?

**Comment by Hjdskes** [comment]
You missed correcting `oftentimes` in `Additional commands to the REPL`.

**Comment by Hjdskes** [comment]
Also, are we going to look into `execution` versus `evaluation`?

File: final/background/spoofax.tex (hunk 6 above)
Comment (Balletie) [inline final/background/spoofax.tex]: it's fine

File: final/background/spoofax.tex (hunk 7 above)
Comment (Balletie) [inline final/background/spoofax.tex]: yes they do that in the paper too.

**Comment by Balletie** [comment]
yes, but I think that is more global than just chapter 2.

File: final/background/spoofax.tex (hunk 5 above)
Comment (Balletie) [inline final/background/spoofax.tex]: it's fine like this

File: final/background/spoofax.tex (hunk 3 above)
Comment (Balletie) [inline final/background/spoofax.tex:55]: the sentence sounds a little weird without it, so I'm keeping it.

**Pull request merged**